x-pack/plugins/search_playground/public/components/app.tsx

```diff
@@ -41,15 +43,18 @@ export const App: React.FC<AppProps> = ({
     }).length
   );
   const handleModeChange = (id: ViewMode) => setSelectedMode(id);
-  const handlePageModeChange = (mode: PlaygroundPageMode) => setSelectedPageMode(mode);
+  const handlePageModeChange = (mode: PlaygroundPageMode) => {
+    services.application?.navigateToUrl(`./${mode}`);
```

Comment:
Wondering if it works in both ES3 and ESS?

Reply:
good catch, I think if we're doing `navigateToUrl` we'd need to do the full URL and not make it relative. 

and if we care doing `navigateToUrl` do we still need the `setSelectedPageMode()` call after this?

Reply:
It does work both places, I tried on both.

Only the last paths differ in between ESS and ES3. When it is relative it works for both. Otherwise I would need to do `isServerless` check and pass full path unless I change some other things completely.

Ideally the routing should be done inside of the plugin completely and ESS wouldn't be able to break it.

Reply:
I will merge it as is for now, and will come back for a refactor to use single router later.